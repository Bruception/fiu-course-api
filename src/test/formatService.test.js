const xml2js = require('xml2js');
const yaml = require('yaml');
const courseDataStore = require('../courseDataStore.js')();
const formatService = require('../formatService.js');

describe('formatService: Testing the formatService module.', () => {
    test('formatService.format: Correctly serializes object to JSON when no format is specified.', () => {
        const courses = courseDataStore.queryBy({
            subject: 'COP',
            isLab: '',
        });
        const formatString = formatService.format(courses);
        expect(JSON.parse(formatString).results).toStrictEqual(courses);
    });
    test('formatService.format: Correctly serializes object to JSON.', () => {
        const courses = courseDataStore.queryBy({
            subject: 'COP',
            isLab: '',
            format: 'json',
        });
        const formatString = formatService.format(courses, 'json');
        expect(JSON.parse(formatString).results).toStrictEqual(courses);
    });
    test('formatService.format: Correctly serializes object to YAML.', () => {
        const courses = courseDataStore.queryBy({
            subject: 'COP',
            isLab: '',
        });
        const formatString = formatService.format(courses, 'yaml');
        expect(yaml.parse(formatString).results).toStrictEqual(courses);
    });
    test('formatService.format: Correctly serializes object to XML.', () => {
        const courses = courseDataStore.queryBy({
            subject: 'COP',
            units: '3.00',
        });
        const formatString = formatService.format(courses, 'xml');
        let parsedCourses;
        xml2js.parseString(formatString,
            {
                explicitArray: false,
            },
            (_err, result) => {
                parsedCourses = result.root;
            }
        );
        expect(courses).toStrictEqual(parsedCourses.results);
        expect(courses.length).toStrictEqual(parseInt(parsedCourses.total));
    });
    test('formatService.format: Correctly serializes object to text.', () => {
        const courses = courseDataStore.queryBy({
            subject: 'COP',
            isLab: '',
        });
        const formatStrings = formatService.format(courses, 'text').split('\n');
        expect(formatStrings[0]).toStrictEqual('total: 1');
        expect(formatStrings[1]).toStrictEqual('subject: COP, code: 2210L, name: Lab for Programming I, units: 0.00, description: Lab component for Computer Programming I Lecture. Corequisite: COP 2210');
    });
});
