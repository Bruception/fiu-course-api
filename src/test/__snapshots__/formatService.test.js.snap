// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatService: Testing the formatService module. formatService.format: Correctly serializes object to text. 1`] = `
"total: 21
results: 
subject: CAP
code: 2752
name: Fundamentals of Data Sci.
units: 3.00
description: This course will teach data science fundamentals to undergraduate non-CS majors. The focus will be on real-world applications and use of associated analysis,  Visualization tools, Python programming. None for B.S. or B.A. standing or permission of the instructor.
subject: CCJ
code: 2010
name: Criminology
units: 3.00
description: Examines the nature, extent, and causes of crime and criminal behavior.
subject: CCJ
code: 2020
name: Criminal Justice
units: 3.00
description: An examination of the history, organization, and function of the criminal justice system, including police, courts, and corrections.
subject: CGN
code: 2161
name: Career Orientation in Civil En
units: 1.00
description: Course provides an overview of the Civil Engineering profession, including understanding of the discipline subfields, to assist students in determining the area(s) of emphasis they might want to follow for their professional career.
subject: CGN
code: 2420
name: Computer Tools for Eng
units: 3.00
description: Introduction to computer software commonly used in engineering, as Excel, Mathcad or Matlab. Numerical techniques to obtain approximated solutions to engineering problems are also included. Prerequisite: PHY2048, and MAC2312 or MAC2282
subject: CGS
code: 2060
name: Intro To Micro Comp
units: 3.00
description: A hands-on study of microcomputer software packages for applications such as operating system, word processing, spreadsheets, and database management. For students without a technical background. Not acceptable for credit for Computer Science majors.
subject: CGS
code: 2100
name: Comp Appls Business
units: 3.00
description: A hands-on study of spreadsheet and database management package for business students without a technical background. Not acceptable for credit for Computer Science majors.
subject: CGS
code: 2518
name: Computr Data Analys
units: 3.00
description: A hands-on study of how to use a modern spreadsheet program to analyze data, including how to perform queries, summarize data, and solve equations.  For non-technical students.  Not acceptable for CS students.
subject: CHI
code: 2200
name: Interm Chinese
units: 3.00
description: Provides intermediate training in the acquisition and application of basic language skills. Prerequisites: One year prior study at college level or permission of the instructor.
subject: CHM
code: 2200
name: Survey Organic Chem
units: 3.00
description: A basic one-semester survey course in organic chemistry for non-majors presenting a broad background in the reactions and structures of organic molecules. Does not fulfill requirements for chemistry, biology, or pre-med majors. Laboratory must be taken concurrently with the course. (Lab fees assessed)
subject: CHM
code: 2200L
name: Survey Org Chem Lab
units: 1.00
description: A basic one-semester survey course in organic chemistry for non-majors presenting a broad background in the reactions and structures of organic molecules. Does not fulfill requirements for chemistry, biology, or pre-med majors. Laboratory must be taken concurrently with the course. (Lab fees assessed)
subject: CHM
code: 2210
name: Organc Chem I
units: 4.00
description: An introduction to chemical bonding and atomic structure theory as it pertains to the chemistry of carbon compounds. Correlation between structure and reactivity of organic molecules followed by a systematic look at the various reaction types using reaction mechanisms as a tool for study. Concurrent registration in both lecture and laboratory is required. (Lab fees assessed)
subject: CHM
code: 2210L
name: Org Chem Lab I
units: 1.00
description: An introduction to chemical bonding and atomic structure theory as it pertains to the chemistry of carbon compounds. Correlation between structure and reactivity of organic molecules followed by a systematic look at the various reaction types using reaction mechanisms as a tool for study. Concurrent registration in both lecture and laboratory is required. (Lab fees assessed)
subject: CHM
code: 2211
name: Organic Chem II
units: 3.00
description: Continuation of CHM 2210, 2210L. Concurrent registration in lecture and laboratory is required. (Lab fees assessed) Lecture is corequisite for lab.
subject: CHM
code: 2211L
name: Organic Chem Lab II
units: 1.00
description: Continuation of CHM 2210, 2210L. Concurrent registration in lecture and laboratory is required. (Lab fees assessed) Lecture is corequisite for lab.
subject: CIV
code: 2222
name: Civic Literacy Test
units: 0.00
description: N/A
subject: CLP
code: 2001
name: Personal Adjustment
units: 3.00
description: Study of personal adjustment in the social and occupational life of the individual. Emphasis on interpersonal aspects of effective behavior.
subject: COP
code: 2210
name: Programming I
units: 4.00
description: A first course in computer science that uses a structured programming language to study programming and problem solving on the computer. Includes the design, construction and analysis of programs.  Student participation in a closed instructional lab is required.  This course will have additional fees.
subject: COP
code: 2250
name: Java Programming
units: 3.00
description: A first course in programming for IT majors.  Syntax and semantics of Java.  Classes and Objects.  Object oriented program development.  Not acceptable for credit for Computer Science majors.  This course will have additional fees.
subject: COP
code: 2270
name: Sec C Prog for Engineers
units: 3.00
description: Secure programming for engineering and science students using ANSI C. Developing algorithms and code for problems in engineering and science, using secure techniques. Not acceptable for CS majors.
subject: CPO
code: 2002
name: Intro Comp Pol
units: 3.00
description: Analysis of major theories of comparative politics including development, state building, institutions, patterns of political interaction and comparative elites. Focus on Latin America and the Third World."
`;

exports[`formatService: Testing the formatService module. formatService.format: Correctly serializes object with no provided options. 1`] = `"{\\"version\\":\\"1.0.0\\",\\"uptime\\":\\"1337\\",\\"status\\":\\"ok\\"}"`;

exports[`formatService: Testing the formatService module. formatService.format: Correctly serializes object with no provided shape function. 1`] = `
"version: 1.0.0
uptime: \\"1337\\"
status: ok
"
`;
